### DISCUSSION TITLE
libcrypt.so.1 is missing

### DISCUSSION THREAD

**Discussion by arossert** [pyinstaller/pyinstaller#8451] [Help]
I'm starting to have an issue with Linux distributions that does not include the `libcrypt.so.1` but instead use `libcrypt.so.2`, I'm getting this error
```
[8086] Error loading Python lib '/tmp/_MEIQoeOR8/libpython3.8.so.1.0': dlopen: libcrypt.so.1: cannot open shared object file: No such file or directory
```
I found this https://github.com/pyinstaller/pyinstaller/issues/6618 and understand that this is intentional, is there a way to work around this and support both versions? since my application is linked to `libcrypt.so.1` so it fails to be executed on systems without it
```
ldd dist/app/_internal/libpython3.8.so.1.0
	linux-vdso.so.1 (0x00007ffd48b2e000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f5e0df4a000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f5e0df11000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f5e0df0c000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f5e0df07000)
	libutil.so.1 => /lib64/libutil.so.1 (0x00007f5e0df02000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f5e0defd000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f5e0de1a000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f5e0d622000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f5e0df6b000)
```

**Comment by rokm** [comment]
The python shared library being linked against `libcrypt.so.1` is an oversight anyway  (only the `_crypt` extension is supposed to be linked against `libcrypt`), and was remedied in later python versions (and builds that use `-Wl,--as-needed`).

But if you are working with python build that is linked against `libcrypt.so.1`, you can explicitly collect the shared library into your application (`--add-binary` or equivalent .spec directive). Although like I said in #6618, it would make more sense for your users to install the compatibility package that provides it.
